### DISCUSSION TITLE
Host Three Js App

### DISCUSSION THREAD

**Discussion by Niris1** [community/community#134285] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

https://github.com/Niris1/NirisTruck

Im playing around with three js and was wandering how do i deploy the app i made..I saw alot of people suggest github pages so I gave it a try..I made my build with vite but I cant get the damn thing to work here on git hub pages...:?

Should i try and host it somewhere else or here should work?

**Comment by DarshanAnand007** [comment]
Hey, you can deploy that in github pages itself , it will work

**Reply by Niris1** [reply]
> Hey, you can deploy that in github pages itself , it will work

well why isnt it working then?Its like the page only has the html file,it doesnt take the folders in consideration..

**Reply by DarshanAnand007** [reply]
go to ur vercel app auth the github and host it
select the prper branch

**Comment by BhathiyaPrasad** [comment]
try vercel.com by linking your account you can import your repo

**Comment by Niris1** [comment]
i moved the main.js file in the root and it is in the sources of the page along with the html file...I guess when i selected the root directory instead of a docs folder it only takes the files that are in the root directory?Is there a tutorial on how to make a custom actions and specify which files to get in the page?

**Reply by BhathiyaPrasad** [reply]
Just try with vercel Brother you can import your GitHub repo to it it will automatically build and deploy your source code 


if this helps you, please mark as answer

**Comment by Niris1** [comment]
no it doesnt because i dont want to use another third party tool..Im already using npm vite ...I moved the pictures and index.js from the assets folder to the root of the directory...Now the page is working displaying the truck and images...But I didnt have to move the files from the textures and models folder..Is there a typo on my end in my html where i included the files or is something else going on?
src="assets/BAR_920_BLACK-gj70QGDs.png" 
src="assets/index.js"

I had the src tags looking like these

**Reply by BhathiyaPrasad** [reply]
Here's how you should update your HTML:


```
<img src="BAR_920_BLACK-gj70QGDs.png" alt="Truck Image">
<script src="index.js"></script>
```
If you still have files in the textures and models folders, you need to ensure that the paths to those files are correct. For example:


```
<img src="textures/texture-file.png" alt="Texture Image">
<script src="models/model-file.js"></script>
```

**Reply by Niris1** [reply]
> Here's how you should update your HTML:
> 
> ```
> <img src="BAR_920_BLACK-gj70QGDs.png" alt="Truck Image">
> <script src="index.js"></script>
> ```
> 
> If you still have files in the textures and models folders, you need to ensure that the paths to those files are correct. For example:
> 
> ```
> <img src="textures/texture-file.png" alt="Texture Image">
> <script src="models/model-file.js"></script>
> ```


Yeah I tell you , I moved the files in the root directory and changed the tags and it is working fine,but why wouldnt it read the files when I had them in an assets folder and tagged them like src="assets/index.js".

using developer tools in chrome i went to sources and all there was there was my index.html...

The tags for the model and texture are working fine getting read from their folders but that is vites work 
 s.load(
      "models/tacoma2024_edited_draco_ultra.glb",
      const e = await s.loadAsync("textures/blouberg_sunrise_2_1k.hdr");
      
     I`ll give you the mark as answer if you try and explain to me.I want to understand for future

**Reply by BhathiyaPrasad** [reply, marked as answer]
Build Process:

Before Deployment: Your development server (Vite) serves files from your project directory, including index.html, assets, and other folders.
After Build: Vite’s build process (npm run build) bundles your application into a dist folder. This process may involve minifying and moving assets to different paths.
GitHub Pages Deployment:

Serving Files: GitHub Pages serves only the contents of the branch or folder you specify (commonly gh-pages branch or docs/ folder).
What’s Included: When you deploy, ensure that all necessary files (including assets and any JavaScript files) are present in the deployable directory (dist).
